src/modeler/AutoRest.Swagger/Validation/AnonymousTypes.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class AnonymousTypes : TypedRule<SwaggerObject>
+    {
+        public override bool IsValid(SwaggerObject entity)
+        {
+            bool valid = true;
+
+            if (entity != null && string.IsNullOrEmpty(entity.Reference))
+            {
+                valid = false;
+            }
+
+            return valid;
+        }
+
+        public override ValidationExceptionName Exception
+        {
+            get
+            {
+                return ValidationExceptionName.AnonymousTypesDiscouraged;
+            }
+        }
```

Comment:
I don't mind if you use c# 6 notation for this sort of thing:

instead of 
``` csharp
        public override ValidationExceptionName Exception
        {
            get
            {
                return ValidationExceptionName.AnonymousTypesDiscouraged;
            }
        }
```

``` csharp
public override ValidationExceptionName Exception => ValidationExceptionName.AnonymousTypesDiscouraged;
```